include/envoy/thread/thread.h

```diff
@@ -7,9 +7,22 @@
 
 #include "common/common/thread_annotations.h"
 
+#if defined(WIN32)
+#include <Windows.h>
+// <windows.h> defines some macros that interfere with our code, so undef them
+#undef DELETE
+#undef GetMessage
+#endif
+
 namespace Envoy {
 namespace Thread {
 
+#if !defined(WIN32)
+using ThreadId = int32_t;
```

Comment:
We can avoid ifdefs in the interface here and instead have ThreadId be an abstract base class, which I'd give two or three methods:
  -- compare to another ThreadID
  -- determine if it's the current thread
  -- serialize to a string for debug purposes

in the past I've done this this way: https://github.com/apache/incubator-pagespeed-mod/blob/20fbc5575c8fc825a137f6f1b2015829a9768f81/pagespeed/kernel/base/thread_system.h#L117

Note referenced there: https://linux.die.net/man/3/pthread_self
```
POSIX.1 allows an implementation wide freedom in choosing the type used to represent a
thread ID; for example, representation using either an arithmetic type or a structure is
permitted. Therefore, variables of type pthread_t can't portably be compared using the
C equality operator (==); use pthread_equal(3) instead.
```